components/organisms/InsightPage/InsightPage.tsx

```diff
@@ -473,18 +499,30 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
             onSearch={(search) => setRepoSearchTerm(search as string)}
           />
 
-          <div>
+          <div className="w-full flex gap-3 md:items-center flex-col md:flex-row">
             <Button
               disabled={
-                repos.length === insightRepoLimit ||
+                repos.length >= insightRepoLimit! ||
                 (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
               }
               loading={addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading}
               onClick={handleAddRepository}
               variant="outline"
+              className="shrink-0 w-max"
             >
               Add Repository
             </Button>
+
+            {repos.length >= insightRepoLimit! && insightRepoLimit! < 50 ? (
+              <span className="text-sm">
+                Your insight pages are limited to
+                <strong className="text-sauced-orange"> {insightRepoLimit}</strong> repos,{" "}
+                <button type="button" onClick={handleUpgrade} className="underline text-sauced-orange">
+                  upgrade
```

Comment:
Instead of doing this, we should link them to the user settings page, where there is an upgrade button

Reply:
This is correct 👍. I was looking at the case where users who are on mobile visit the settings and they don't get to see the upgrade button in first view... but i have a walk around for this 👍